src/uct/tcp/tcp_ep.c

```diff
@@ -600,24 +601,27 @@ static void uct_tcp_iface_check_rp_filter(uct_tcp_iface_t *iface)
     ucs_status_t status;
 
     if (iface_saddr->sa_family != AF_INET) {
-        return;
+        status = UCS_OK;
+        goto out;
     }
 
     ucs_snprintf_safe(rp_filter_path, MAXPATHLEN,
                       "/proc/sys/net/ipv4/conf/%s/rp_filter", iface->if_name);
     status = ucs_read_file_number(&rp_filter, 1, "%s", rp_filter_path);
     if (status != UCS_OK) {
-        ucs_diag("tcp_iface %p: unable to read rp_filter from %s", iface,
-                 rp_filter_path);
-        return;
+        ucs_log(log_level,"tcp_iface %p: unable to read rp_filter from %s",
+                iface, rp_filter_path);
+        goto out;
     }
 
     if (rp_filter == 1) {
-        ucs_diag("tcp_iface %p: rp_filter of %s is set to strict mode, "
-                 "it may cause packet check failures, see "
-                 "https://www.kernel.org/doc/Documentation/networking/ip-sysctl.txt"
-                 " for details", iface, iface->if_name);
+        ucs_log(log_level, "tcp_iface %p: net.ipv4.conf.%s.rp_filter is set to "
+                "strict mode (1), connections may fail", iface, iface->if_name);
+        status = UCS_ERR_INVALID_PARAM;
     }
+
+out:
+    return (log_level <= UCS_LOG_LEVEL_ERROR) ? status : UCS_OK;
```

Comment:
this logic should be outside the function, and according to the config ternary value